<h1>set_terminate</h1>
<p><pre class="codehilite"><code class="language-cpp"><span style="color: #000">namespace std {
  typedef void (*terminate_handler)();
  terminate_handler set_terminate(terminate_handler f) noexcept;
}</span></code></pre>
</p>
<h2>概要</h2>
<p>例外をキャッチされなかった場合に実行される終了ハンドラを設定する。</p>
<h2>戻り値</h2>
<p>前に設定されていた終了ハンドラを返す。</p>
<h2>例外</h2>
<p>投げない</p>
<h2>備考</h2>
<p>引数としてヌルポインタが渡された場合、デフォルトの終了ハンドラが設定される。</p>
<p>終了ハンドラは、呼び出し元に戻らずに異常終了すべきと規定されている。</p>
<h2>例</h2>
<p><pre class="codehilite"><code class="language-cpp"><span style="color:#ff0000">set_terminate</span>(my_terminate_handler);

  throw std::exception(); // no catch
}<span style="color: #000">#include &lt;iostream&gt;
#include &lt;exception&gt;

void my_terminate_handler()
{
  std::cout &lt;&lt; "my terminate handler" &lt;&lt; std::endl;
}

int main()
{
  std::</span></code></pre>
</p>
<h3>出力例</h3>
<p><pre><code><span style="color: #000">my terminate handler

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
</span></code></pre></p>
<h2>参照</h2>
<footer><a href="https://github.com/cpprefjp/site/edit/master/reference/exception/set_terminate.md" target="_blank">編集</a></footer>